src/rule-expression-language.ts

```diff
@@ -12,13 +12,94 @@ import {
   TermType,
 } from 'expressionparser/dist/ExpressionParser.js';
 import get from 'lodash/get.js';
+import set from 'lodash/set.js';
+import isNumberLodash from 'lodash/isNumber.js';
 import ms from 'ms';
+import { isNumber } from './utils/isNumber';
 import { toArray } from './utils/toArray';
 
 export interface FunctionOps {
   [op: string]: (...args: ExpressionThunk[]) => ExpressionValue;
 }
 
+export const assignmentOperators = ['+=', '='];
+// TODO: , '-=', '*=', '/=', '%='];
+/*
+'+='
+'-='
+'*='
+'/='
+'**='
+'%='
+'||='
+'??='
+*/
+
+const getInfixOps = (termDelegate: TermDelegate): InfixOps => ({
+  '+': (a, b) => num(a()) + num(b()),
+  '-': (a, b) => num(a()) - num(b()),
+  '*': (a, b) => num(a()) * num(b()),
+  '/': (a, b) => num(a()) / num(b()),
+  ',': (a, b): ArgumentsArray => {
+    const aVal = a();
+    const aArr: ExpressionArray<ExpressionValue> = isArgumentsArray(aVal)
+      ? aVal
+      : [() => aVal];
+    const args: ExpressionArray<ExpressionValue> = aArr.concat([b]);
+    args.isArgumentsArray = true;
+    return args as ArgumentsArray;
+  },
+  '%': (a, b) => num(a()) % num(b()),
+  '+=': (a, b) => {
+    const token = a();
+    const value = b();
+    const resolvedToken =
+      typeof token === 'number'
+        ? token
+        : typeof token === 'string'
+        ? termDelegate(token)
+        : (undefined as never);
+    const resolvedValue =
+      typeof value === 'number' && isFinite(value)
+        ? value
+        : typeof value === 'string'
+        ? termDelegate(value)
+        : NaN;
+
+    if (isNumberLodash(resolvedToken) && isNumberLodash(resolvedValue)) {
+      return resolvedToken + resolvedValue;
+    } else {
+      throw new Error(
+        `Cannot apply += operation to expression: ${token} += ${value}`
+      );
+    }
+  },
+  '=': (_, b) => {
+    const value = b();
+    const resolvedValue =
+      typeof value === 'number' && isFinite(value)
+        ? value
+        : typeof value === 'string'
+        ? termDelegate(value)
+        : NaN;
+
+    return value;
```

Comment:
Just using value here because when is a string `resolvedValue` ends up trying to interpret it as a variable. In general how can we differentiate between a variable name and a value string here?

Reply:
In this part of the code, I don't think it sees the string token.

(If it does, I don't think we should be doing the label-to-value lookups as it should already be done in another function - the one which uses `termDelegate` in the switch statement.)

Do we know `termDelegate` gets hit here?